src/ui/mainwin.cpp

```diff
@@ -3478,47 +3479,47 @@ void CAMainWin::on_uiImportDocument_triggered()
 
         CACanorus::rebuildUI(document());
     } else {
-        CAImport* import = nullptr;
+        if (_importFile) {
+            _importFile.reset();
+        }
 
         if (uiImportDialog->selectedNameFilter() == CAFileFormats::MIDI_FILTER) {
             if (!document())
                 newDocument();
-            /// \todo replace raw pointer with shared or unique pointer
-            import = new CAMidiImport(document());
-            if (import) {
-                import->setStreamFromFile(s);
-                connect(import, SIGNAL(importDone(int)), this, SLOT(onImportDone(int)));
-                import->importSheet();
+            _importFile = std::make_unique<CAMidiImport>();
+            if (_importFile) {
+                _importFile->setStreamFromFile(s);
+                connect(_importFile.get(), SIGNAL(importDone(int)), this, SLOT(onImportDone(int)));
+                _importFile->importSheet();
             }
         } else if (uiImportDialog->selectedNameFilter() == CAFileFormats::LILYPOND_FILTER) {
             // activate this filter in src/canorus.cpp when sheet import is usable
             if (!document())
                 newDocument();
-            /// \todo replace raw pointer with shared or unique pointer
-            import = new CALilyPondImport(document());
-            if (import) {
-                import->setStreamFromFile(s);
-                connect(import, SIGNAL(importDone(int)), this, SLOT(onImportDone(int)));
-                import->importSheet();
+            _importFile = std::make_unique<CALilyPondImport>();
```

Comment:
Forgot to pass document() arg to constructor?

Reply:
Done. Prepared for next comit. Thanks